nixos/lib/make-options-doc/mergeJSON.py

```diff
@@ -41,6 +41,138 @@ def unpivot(options: Dict[Key, Option]) -> Dict[str, JSON]:
         result[opt.name] = opt.value
     return result
 
+# converts in-place!
+def convertMD(options: Dict[str, Any]) -> str:
+    import mistune
+    import re
+    from xml.sax.saxutils import escape, quoteattr
+
+    admonitions = {
+        '.warning': 'warning'
+    }
+    class Renderer(mistune.renderers.BaseRenderer):
+        def _get_method(self, name):
+            try:
+                return super(Renderer, self)._get_method(name)
+            except AttributeError:
+                def not_supported(children, **kwargs):
+                    raise NotImplementedError("md node not supported yet", name, children, **kwargs)
+                return not_supported
+
+        def text(self, text):
+            return escape(text)
+        def paragraph(self, text):
+            return text + "\n\n"
```

Comment:
Should not this be wrapped in `<para>`, example: https://nixos.org/manual/nixos/stable/options.html#opt-services.matrix-synapse.settings

Reply:
the `<para>` is generated by the container, eg in line 88. the way this script uses `paragraph` is more akin to paragraph *breaks*, because that best mimics how the current system treats the double newline as a paragraph breaks.

Reply:
That would not work for the outermost paragraphs though.

----

On a different note, I wonder if we should just do preprocessing here and skip https://github.com/NixOS/nixpkgs/blob/e0e9109c12eaee527fbae2e7ac3d6b79588b26d3/nixos/lib/make-options-doc/postprocess-option-descriptions.xsl. I can’t imagine that to be very fast.

Reply:
not sure we understand what you mean. there aren't any "outer" paragraphs as such, the docbook `description` is pasted into a `<para>` element in https://github.com/NixOS/nixpkgs/blob/e0e9109c12eaee527fbae2e7ac3d6b79588b26d3/nixos/lib/make-options-doc/options-to-docbook.xsl#L55. the postprocessing you linked artifically splits that wrapper parapgraph at markdown paragraph boundaries by turning the description into a technically-invalid xml fragment.

as for skipping the preprocessing: don't think that's feasible unless we convert *all* options that use markdown paragraph separation at the same time as well.

Reply:
I meant skipping preprocessing for the markdown content hoping for better performance as less and less DocBook descriptions exist. But it appears the performance impact is negligible anyway.